akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -986,32 +995,37 @@ private[akka] class ShardRegion(
   def registrationMessage: Any =
     if (entityProps.isDefined) Register(self) else RegisterProxy(self)
 
-  def requestShardBufferHomes(): Unit = {
-    // Have to use vars because MessageBufferMap has no map, only foreach
-    var totalBuffered = 0
-    var shards = List.empty[String]
-    shardBuffers.foreach {
-      case (shard, buf) =>
-        coordinator.foreach { c =>
-          totalBuffered += buf.size
-          shards ::= shard
-          log.debug(
-            "{}: Retry request for shard [{}] homes from coordinator at [{}]. [{}] buffered messages.",
-            typeName,
-            shard,
-            c,
-            buf.size)
-          c ! GetShardHome(shard)
-        }
-    }
+  /**
+   * Send GetShardHome for all shards with buffered messages
+   * If coordinator is empty, nothing happens
+   */
+  def tryRequestShardBufferHomes(): Unit = {
+
+    coordinator.foreach { coord =>
+      // Have to use vars because MessageBufferMap has no map, only foreach
+      var totalBuffered = 0
+      var shards = List.empty[String]
+      shardBuffers.foreach {
+        case (shard, buf) =>
+            totalBuffered += buf.size
+            shards ::= shard
+            log.debug(
+              "{}: Retry request for shard [{}] homes from coordinator at [{}]. [{}] buffered messages.",
+              typeName,
+              shard,
+              coord,
+              buf.size)
+          coord ! GetShardHome(shard)
+      }
 
-    if (retryCount >= 5 && retryCount % 5 == 0 && log.isWarningEnabled) {
-      log.warning(
-        "{}: Retry request for shards [{}] homes from coordinator. [{}] total buffered messages. Coordinator [{}]",
-        typeName,
-        shards.sorted.mkString(","),
-        totalBuffered,
-        coordinator)
+      if (retryCount >= 5 && retryCount % 5 == 0 && log.isWarningEnabled) {
+        log.warning(
+          "{}: Retry request for shards [{}] homes from coordinator. [{}] total buffered messages. Coordinator [{}]",
+          typeName,
+          shards.sorted.mkString(","),
+          totalBuffered,
+          coord)
+      }
```

Comment:
Here the difference is that if `coordinator` is empty nothing happens. That was already the case, but the warning log was run with or without coordinator. 

It says that something happens, but if `coordinator.isEmpty` nothing happens.

Reply:
I think it was useful to have that logging also when coordinator was None, because that can happen for example if coordinator node crashed but wasn't downed/removed. Then it's good to see that something is stalled and buffer size increases.

Reply:
Ok, I will revert this part

Reply:
@patriknw, I was looking into this in detail. The method `requestShardBufferHomes` (now renamed to `tryRequestShardBufferHomes` is only called twice here. 

It's called when we get the `RegisterAck` in which case we have the coordinator. 

And it's called when we receive `Retry`, but only if the coordinator is not empty. 

Moving everything inside `coordinator.foreach` makes sense in that case. 

However, we could add a similar logging whenever get a `Retry` and we see that coordinator is `None`. Then we can log how many message we have pending.

Reply:
btw, when coordinator is `None` in the previous code, `totalBuffered` and `shards` are not updated, so we log that we have zero shards and zero messages.

Reply:
btw 2

`register()` is already logging whenever we have buffered messages. So I think that extra logging we are talking here is unnecessary.